DFC.App.FindACourse/Controllers/DetailsController.cs

```diff
@@ -3,26 +3,28 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Logging;
 using System;
+using System.Threading.Tasks;
 
 namespace DFC.App.FindACourse.Controllers
 {
     public class DetailsController : Controller
     {
-        private readonly ILogger<CourseController> logger;
+        private readonly ILogger<DetailsController> logger;
         private readonly IFindACourseService findACourseService;
 
-        public DetailsController(ILogger<CourseController> logger, IFindACourseService findACourseService)
+        public DetailsController(ILogger<DetailsController> logger, IFindACourseService findACourseService)
         {
             this.logger = logger;
             this.findACourseService = findACourseService;
         }
 
         [HttpGet]
         [Route("find-a-course/course/body/course/details")]
-        public IActionResult Details(string courseId, string runId)
+        public async Task<IActionResult> Details(string courseId, string runId, string searchTerm)
         {
             this.logger.LogInformation($"{nameof(this.Details)} has been called");
             var model = new DetailsViewModel();
+            model.SearchTerm = searchTerm;
 
             if (string.IsNullOrEmpty(courseId) || string.IsNullOrEmpty(runId))
             {
```

Comment:
When I looked at this file its throwing a generic exception, can you check that please think in need to be a more specific one.

Reply:
Changed